src/inventory.cpp

```diff
@@ -3,9 +3,7 @@
 
 #include "inventory.hpp"
 
-#include "config.hpp"
-
-#include <sdbusplus/bus.hpp>
+#include <config.hpp>
```

Comment:
This file is created by meson in a build dir, it is not a local file. Meson adds this dir with `-I` options for GCC. Clang format reorders include according to own logic, I can't change this behavior.

Reply:
This is a project specific file not a system specific. So, it should be in quotes in my opinion.
About order: you could rename it to `config.h` or change the rule at the line 55 of `.clang-format`:
```YAML
  - Regex:           '^"config.h"'
    Priority:        -1
```

Reply:
Quotes doesn't define the type of file (system or  project), it is a rule for a compiler to construct order for a file searching oprtation. In our case the file `config.h` is not placed to the directory of source files (`src`), so using `"` is wrong.

Reply:
This file is generated  during the build process.
It means that it is a part of the project sources and it should not be searched in the system headers. ;)

Reply:
https://gcc.gnu.org/onlinedocs/cpp/Search-Path.html
`By default, the preprocessor looks for header files included by the quote form of the directive #include "file" first relative to the directory of the current file`

The file `config.h` is not in the same dir (where `inventory.cpp` is). So, the compiler will look into system includes anyway. Using `<>` tells the compiler to skip the current directory from search path.

Reply:
Yes, It seems that for meson, it would be more correct to add the build directory with `-iqoute` option but they do it with a more universal `-I`.

But you as a human, where would you look for the `<some-unknown-header.hpp>` firstly? And `"some-unknown-header.hpp"`? :)

Ok, I don't insist. If you like more the angle brackets let's leave it as is.

Reply:
I agree with @nest1ing on this. It would be better to have it in double quotes.

Reply:
Even Meson's [documentation](https://mesonbuild.com/Configuration.html) suggests using `<>`.
I created the [PR](https://gerrit.openbmc-project.xyz/c/openbmc/docs/+/32742) for clang-format, let's see what they say.

Reply:
Ok, you win ))
I was sure it is in C++ standard, but standard says that brackets behavior is "implementation-defined".